src/engine/sparqlExpressions/StringExpressions.cpp

```diff
@@ -361,6 +361,33 @@ using StrBeforeExpression =
     StringExpressionImpl<2, LiftStringFunction<decltype(strBefore)>,
                          StringValueGetter>;
 
+[[maybe_unused]] auto mergeFlagsIntoRegex =
+    [](std::optional<std::string> regex,
+       const std::optional<std::string>& flags) -> IdOrLiteralOrIri {
+  if (!flags.has_value() || !regex.has_value()) {
+    return Id::makeUndefined();
+  }
+  auto firstInvalidFlag = flags.value().find_first_not_of("imsu");
+  if (firstInvalidFlag != std::string::npos) {
+    throw std::runtime_error{absl::StrCat(
+        "Invalid regex flag '",
+        std::string_view{&flags.value()[firstInvalidFlag], 1}, "' found in \"",
+        flags.value(),
+        "\". The only supported flags are 'i', 'm', 's', 'u', and any "
+        "combination of them")};
+  }
```

Comment:
The error case in sparql expressions is not a hard exception but
has to be handled via `Id::makeUndefined` or something else.

Reply:
Yes, it's currently partially duplicated. But I'm planning to unify the code from both classes so the duplicate code will disappear soon.